plugins/account_history_plugin/include/eos/account_history_plugin/account_history_object.hpp

```diff
@@ -13,15 +26,18 @@ class transaction_history_object : public chainbase::object<chain::transaction_h
    id_type             id;
    block_id_type       block_id;
    transaction_id_type transaction_id;
+   AccountName         account_name;
 };
 
 struct by_id;
 struct by_trx_id;
+struct by_account_name;
 using transaction_history_multi_index = chainbase::shared_multi_index_container<
    transaction_history_object,
    indexed_by<
       ordered_unique<tag<by_id>, BOOST_MULTI_INDEX_MEMBER(transaction_history_object, transaction_history_object::id_type, id)>,
-      hashed_unique<tag<by_trx_id>, BOOST_MULTI_INDEX_MEMBER(transaction_history_object, transaction_id_type, transaction_id), std::hash<transaction_id_type>>
+      hashed_non_unique<tag<by_trx_id>, BOOST_MULTI_INDEX_MEMBER(transaction_history_object, transaction_id_type, transaction_id), std::hash<transaction_id_type>>,
+      hashed_non_unique<tag<by_account_name>, BOOST_MULTI_INDEX_MEMBER(transaction_history_object, AccountName, account_name), std::hash<AccountName>>
```

Comment:
Fwiw, we avoid non unique indexes in core validation, because they're inconsistent across machines. This isn't core validation, it's just an API call, so it shouldn't matter so much here.